vhost-device-sound/src/audio_backends/alsa.rs

```diff
@@ -196,21 +196,21 @@ fn write_samples_direct(
     mmap: &mut alsa::direct::pcm::MmapPlayback<u8>,
 ) -> AResult<bool> {
     while mmap.avail() > 0 {
-        let Some(buffer) = stream.buffers.front_mut() else {
+        let Some(request) = stream.requests.front_mut() else {
```

Comment:
The renaming merged with the other refactoring makes it a bit harder than necessary to review the actual functional changes. IMHO it is still manageable here, but for the future you might want to consider doing that in a separate commit.

Reply:
I agree. The reason I used buffer instead of request was that, with this commit, the entire request is enqueued instead of individual buffers. I thought the variable name should reflect the changed meaning of the queue elements. Do you think it would be better to keep buffer?

Reply:
IMHO "request" is better.
Maybe you can do the rename in a next patch, but I'm not sure about it.